src/test/java/org/folio/linked/data/validation/dto/LccnPatternValidatorTest.java

```diff
@@ -4,90 +4,115 @@
+      arguments("  n12345678 ", "http://id.loc.gov/vocabulary/mstatus/current", List.of(createSpecRule(true))),
+      arguments("   1234567 ", "http://id.loc.gov/vocabulary/mstatus/current", List.of(createSpecRule(true))),
+      arguments("   12345678", "http://id.loc.gov/vocabulary/mstatus/current", List.of(createSpecRule(true))),
+      arguments(" nnn123456789 ", "http://id.loc.gov/vocabulary/mstatus/current", List.of(createSpecRule(true))),
+      arguments("nnn123456789 ", "http://id.loc.gov/vocabulary/mstatus/current", List.of(createSpecRule(true))),
+      arguments("nnnn12345678 ", "http://id.loc.gov/vocabulary/mstatus/current", List.of(createSpecRule(true))),
+      arguments("nNn12345678 ", "http://id.loc.gov/vocabulary/mstatus/current", List.of(createSpecRule(true))),
+      arguments("nNn12345678 ", null, List.of(createSpecRule(true))),
+      arguments("n-n12345678 ", "http://id.loc.gov/vocabulary/mstatus/current", List.of(createSpecRule(true))),
 
       // structure B
-      arguments(" 0123456789", "http://id.loc.gov/vocabulary/mstatus/current"),
-      arguments(" m123456789", "http://id.loc.gov/vocabulary/mstatus/current"),
-      arguments("  m123456789 ", "http://id.loc.gov/vocabulary/mstatus/current"),
-      arguments(" mm123456789 ", "http://id.loc.gov/vocabulary/mstatus/current"),
-      arguments("mm123456789 ", "http://id.loc.gov/vocabulary/mstatus/current"),
-      arguments("mmm0123456789", "http://id.loc.gov/vocabulary/mstatus/current"),
-      arguments("nN0123456789", "http://id.loc.gov/vocabulary/mstatus/current"),
-      arguments("nN0123456789", null),
-      arguments("n-0123456789", "http://id.loc.gov/vocabulary/mstatus/current"));
+      arguments(" 0123456789", "http://id.loc.gov/vocabulary/mstatus/current", List.of(createSpecRule(true))),
+      arguments(" m123456789", "http://id.loc.gov/vocabulary/mstatus/current", List.of(createSpecRule(true))),
+      arguments("  m123456789 ", "http://id.loc.gov/vocabulary/mstatus/current", List.of(createSpecRule(true))),
+      arguments(" mm123456789 ", "http://id.loc.gov/vocabulary/mstatus/current", List.of(createSpecRule(true))),
+      arguments("mm123456789 ", "http://id.loc.gov/vocabulary/mstatus/current", List.of(createSpecRule(true))),
+      arguments("mmm0123456789", "http://id.loc.gov/vocabulary/mstatus/current", List.of(createSpecRule(true))),
+      arguments("nN0123456789", "http://id.loc.gov/vocabulary/mstatus/current", List.of(createSpecRule(true))),
+      arguments("nN0123456789", null, List.of(createSpecRule(true))),
+      arguments("n-0123456789", "http://id.loc.gov/vocabulary/mstatus/current", List.of(createSpecRule(true))));
   }
 
   private LccnRequest createLccnRequest(String value, String link) {
     return new LccnRequest()
       .value(List.of(value))
       .status(link == null ? emptyList() : List.of(new Status().link(List.of(link))));
   }
+
+  private static SpecificationRuleDto createSpecRule(boolean enabled) {
```

Comment:
we can return List here, so no need to explicitly wrap spec rule each time in arguments

Reply:
done